tokio-util/src/sync/cancellation_token.rs

```diff
@@ -241,6 +241,52 @@ impl CancellationToken {
     pub fn drop_guard(self) -> DropGuard {
         DropGuard { inner: Some(self) }
     }
+
+    /// Runs a future to completion and returns its result wrapped inside of an `Option`
+    /// unless the `CancellationToken` is cancelled. In that case the function returns
+    /// `None` and the future gets dropped.
+    ///
+    /// # Cancel safety
+    ///
+    /// This method is not cancel safe since the input future gets dropped on cancel.
```

Comment:
I think we can be a bit more precise here.
```suggestion
    /// This method is only cancel safe if `fut` is cancel safe.
```